components/selection/selection-observer-mixin.js

```diff
@@ -15,68 +15,36 @@ export const SelectionObserverMixin = superclass => class extends superclass {
 			 * @ignore
 			 * @type {object}
 			 */
-			selectionInfo: { type: Object },
-			_provider: { type: Object, attribute: false }
+			selectionInfo: { type: Object }
 		};
 	}
 
 	constructor() {
 		super();
 		this.selectionInfo = new SelectionInfo();
-		this._provider = null;
+
+		this._subscriberController = new SubscriberController(this,
+			{ onUnsubscribe: this._clearSelectionInfo.bind(this) },
+			{ eventName: 'd2l-selection-observer-subscribe', forProperty: 'selectionFor', controllerId: 'observer' }
+		);
 	}
 
 	connectedCallback() {
 		super.connectedCallback();
-		// delay subscription otherwise import/upgrade order can cause selection mixin to miss event
-		requestAnimationFrame(() => {
-			if (this.selectionFor) return;
-
-			const evt = new CustomEvent('d2l-selection-observer-subscribe', {
-				bubbles: true,
-				composed: true,
-				detail: {}
-			});
-			this.dispatchEvent(evt);
-			this._provider = evt.detail.provider;
-		});
+		this._subscriberController.hostConnected();
```

Comment:
This is a good sign -- once we go to Lit 2, `connectedCallback`, `disconnectedCallback` and `updated` can be removed.